frontend/src/store/wallet.ts

```diff
@@ -362,22 +369,41 @@ export default function useWalletStore() {
   // ------------------------------------- Computed parameters -------------------------------------
   // "True" computed properties, i.e. derived from this module's state
 
-  const currentChain = computed(() => {
-    return getChainById(network.value?.chainId || 1);
-  });
+  const chainId = computed(() => network.value?.chainId); // returns a number
+  const currentChain = computed(() => getChainById(chainId.value || 1));
 
   const isSupportedNetwork = computed(() => {
-    if (!network.value) return true; // assume valid if we have no network information
-    return supportedChainIds.includes(network.value.chainId);
+    if (!network.value || !chainId.value) return true; // assume valid if we have no network information
+    return supportedChainIds.includes(chainId.value);
   });
 
+  const getNativeTokenMinSend = (chainId: number): string => {
+    switch (chainId) {
+      case 137:
+        return parseUnits('0.15', 'ether').toString(); // Polygon
+      default:
+        return parseUnits('0.02', 'ether').toString(); // everything else
+    }
+  };
+
   const NATIVE_TOKEN = computed(() => {
-    return { ...(currentChain.value?.nativeCurrency as TokenInfo), chainId: network.value?.chainId as number };
+    // this value is used if the relayer is down
+    const fallbackMinSend = getNativeTokenMinSend(chainId.value!);
+    return {
+      ...(currentChain.value!.nativeCurrency as TokenInfoExtended),
+      minSendAmount: relayer.value?.nativeTokenMinSendAmount || fallbackMinSend,
+      chainId: chainId.value!,
+    };
   });
 
-  const tokens = computed((): TokenInfo[] => {
-    // Add ETH as a supported token
+  const tokens = computed((): TokenInfoExtended[] => {
     const supportedTokens = relayer.value?.tokens || [];
+    const isNativeTokenInRelayerTokenList = supportedTokens
+      .map((token) => token.address)
+      .includes(NATIVE_TOKEN.value.address);
```

Comment:
Can we make sure the native token is always first in this list? Since I believe this order is the order used in the token dropdown. I think native token first, sort alphabetically by symbol for the rest would be great.